### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by IvanFadieiev**

### CODE REVIEW

File: src/main/java/project/bookstore/repository/impl/BookRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                throw new RuntimeException("Can't get book " + book.getTitle() + " from DB");
+            }
+            transaction.rollback();
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return book;
+    }
```
Comment (AntonZhdanov) [inline src/main/java/project/bookstore/repository/impl/BookRepositoryImpl.java:42]: Maybe it's better to up return block in try section? I think it's best practice. But this is my recommendation.

File: src/main/java/project/bookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query query = session.createQuery("FROM Book", Book.class);
+            return query.getResultList();
+        } catch (Throwable e) {
+            throw new RuntimeException("Can't get all books from DB");
+        }
+    }
+}
```
Comment (AntonZhdanov) [inline src/main/java/project/bookstore/repository/impl/BookRepositoryImpl.java:53]: add empty line.

File: src/main/java/project/bookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    @Override
+    public Book save(Book book) {
+        bookRepository.save(book);
+        return book;
```
Comment (AntonZhdanov) [inline src/main/java/project/bookstore/service/impl/BookServiceImpl.java:22]: in this case maybe better return booRepository.save(book)?

**Review by AntonZhdanov — APPROVED** [review]

**Comment by AntonZhdanov** [comment]
Good solution. Fix some checkstyle mistakes if you want.

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**